igor-core/src/main/java/com/netflix/spinnaker/igor/build/model/GenericBuild.java

```diff
@@ -49,6 +49,21 @@
 
   private Map<String, ?> properties;
 
+  public GenericBuild withGitInfo(List<GenericGitRevision> revs) {
+    genericGitRevisions = revs;
+    return this;
+  }
+
+  public GenericBuild withProperties(Map<String, ?> properties) {
+    this.properties = properties;
+    return this;
+  }
+
+  public GenericBuild withArtifacts(List<GenericArtifact> artifacts) {
+    this.artifacts = artifacts;
+    return this;
+  }
```

Comment:
IMHO these methods should be removed and instead added using the Lombok annotation `@Wither`. You can probably just add it for the whole class, but if you only want these three methods, you can add it directly on the corresponding fields.

Reply:
added `@Wither` to the whole class, doesn't seem to hurt. thanks!